Drivers/Periphs/Src/SPI.c

```diff
@@ -0,0 +1,84 @@
+/* USER CODE BEGIN Header */
+/**
+  ******************************************************************************
+  * @file    spi.c
+  * @brief   This file provides code for the configuration
+  *          of all used SPI pins.
+  ******************************************************************************
+  * @attention
+  *
+  * Copyright (c) 2023 STMicroelectronics.
+  * All rights reserved.
+  *
+  * This software is licensed under terms that can be found in the LICENSE file
+  * in the root directory of this software component.
+  * If no LICENSE file comes with this software, it is provided AS-IS.
+  *
+  ******************************************************************************
+  */
+/* USER CODE END Header */
+
+/* Includes ------------------------------------------------------------------*/
+#include "SPI.h"
+
+/* USER CODE BEGIN 0 */
+
+/* USER CODE END 0 */
+
+SPI_HandleTypeDef hspi1;
+SPI_HandleTypeDef hspi2;
+
+/*----------------------------------------------------------------------------*/
+/* Configure SPI                                                              */
+/*----------------------------------------------------------------------------*/
+/* USER CODE BEGIN 1 */
+
+/**
+  * @brief SPI1 Initialization Function
+  * @param port Which port to initialize (1 or 2)
+  * @retval None
+  */
+void MX_SPI_Init(SPI_TypeDef* port){
+    if(port = SPI1){
+        hspi1.Instance = SPI1;
+        hspi1.Init.Mode = SPI_MODE_MASTER;
+        hspi1.Init.Direction = SPI_DIRECTION_2LINES;
+        hspi1.Init.DataSize = SPI_DATASIZE_8BIT;
+        hspi1.Init.CLKPolarity = SPI_POLARITY_LOW;
+        hspi1.Init.CLKPhase = SPI_PHASE_1EDGE;
+        hspi1.Init.NSS = SPI_NSS_HARD_OUTPUT;
+        hspi1.Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_128; //start slow then increase when using
+        hspi1.Init.FirstBit = SPI_FIRSTBIT_MSB;
+        hspi1.Init.TIMode = SPI_TIMODE_DISABLE;
+        hspi1.Init.CRCCalculation = SPI_CRCCALCULATION_DISABLE;
+        hspi1.Init.CRCPolynomial = 10;
+        if (HAL_SPI_Init(&hspi1) != HAL_OK)
+        {
+            Error_Handler();
+        }
+    }else if(port = SPI2){
+        hspi2.Instance = SPI2;
+        hspi2.Init.Mode = SPI_MODE_MASTER;
+        hspi2.Init.Direction = SPI_DIRECTION_2LINES;
+        hspi2.Init.DataSize = SPI_DATASIZE_8BIT;
+        hspi2.Init.CLKPolarity = SPI_POLARITY_LOW;
+        hspi2.Init.CLKPhase = SPI_PHASE_1EDGE;
+        hspi2.Init.NSS = SPI_NSS_HARD_OUTPUT;
```

Comment:
note that SPI2 is not connected to anything on the board except for SPI2_NSS which overlaps with our SPI1 control signal pin. Ideally MX_SPI_Init(hspi2) should never be called, but you should set hspi2.Init.NSS to SPI_NSS_SOFT just in case to avoid interference.